src/components/breadcrumb/breadcrumb.js

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Icon from '@mapbox/mr-ui/icon';
+import { createUniqueCrumbs } from './utils';
+
+export default class Breadcrumb extends React.Component {
+  render() {
+    const { currentPage, section, domain, site, subsite } = this.props;
+
+    const Link = (props) => (
+      <React.Fragment>
+        <a className="link" href={props.href}>
+          {props.children}
+        </a>
+        <span className="color-gray-light inline-block px6">
+          <Icon name="chevron-right" inline={true} />
+        </span>
+      </React.Fragment>
+    );
+
+    const createLinks = createUniqueCrumbs([
+      ...(domain !== false ? [domain] : []),
+      ...(site ? [site] : []),
+      ...(subsite ? [subsite] : []),
+      ...(section ? [section] : []),
+      ...(currentPage ? [currentPage] : [])
+    ]);
+
+    return createLinks.length > 1 ? (
+      <div
+        className="dr-ui--breadcrumb none block-mm px24 pt12"
+        data-swiftype-index="false"
+      >
+        {createLinks.map((link) => {
+          return link.path !== currentPage.path ? (
+            <Link key={link.title} href={link.path}>
+              {link.title}
+            </Link>
+          ) : (
+            <span className="color-gray">{link.title}</span>
+          );
+        })}
+      </div>
+    ) : null;
+  }
+}
+
+Breadcrumb.propTypes = {
+  /** If `false`, do not show "All docs" as first breadcrumb link */
+  domain: PropTypes.oneOfType([
+    PropTypes.shape({
+      title: PropTypes.string.isRequired,
+      path: PropTypes.string.isRequired
+    }),
+    PropTypes.bool
+  ]),
+  /** Title of the site and homepage path */
+  site: PropTypes.shape({
+    title: PropTypes.string.isRequired,
+    path: PropTypes.string.isRequired
+  }),
+  /** Subsite title and homepage path (if using multi-structured layout) */
```

Comment:
`subsite` and `section` just dictate the order of the breadcrumbs, right? Are there any scenarios where we might want multi-level breadcrumbs for something that doesn't have a true subsite?

Reply:
> subsite and section just dictate the order of the breadcrumbs, right? 

Correct

> Are there any scenarios where we might want multi-level breadcrumbs for something that doesn't have a true subsite?

I'm unsure of a scenario right now, but if we find one we can add a test case and fix to cover it.

Reply:
The new array setup looks really great to me!